ingestclient/test/test_backend_join_retry.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2016 The Johns Hopkins University Applied Physics Laboratory
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import
+from ingestclient.core.backend import BossBackend, Backend
+from ingestclient.test.aws import Setup
+
+import boto3
+import os
+import unittest
+import json
+import responses
+from pkg_resources import resource_filename
+import six
+import mock
+
+ERROR_TEXT = "Error on the server"
+
+class ResponsesMixin(object):
+    """Mixin to setup requests mocking for the test class"""
+    def setUp(self):
+        responses._default_mock.__enter__()
+        self.add_default_response()
+        super(ResponsesMixin, self).setUp()
+
+    def tearDown(self):
+        super(ResponsesMixin, self).tearDown()
+        responses._default_mock.__exit__(None, None, None)
+
+    def add_default_response(self):
+        mocked_repsonse = {"id": 23}
+
+        mocked_repsonse = {"text": ERROR_TEXT}
+        responses.add(responses.GET, 'https://api.theboss.io/latest/ingest/23',
+                      json=mocked_repsonse, status=500)
+
+
+
+
+class BossBackendTestMixin(object):
+
+
+    @mock.patch('time.sleep')
+    def test_join_retry(self, fake_sleep):
+        """Test creating an ingest job - mock server response"""
+        b = BossBackend(self.example_config_data)
+        b.setup(self.api_token)
+
+        with self.assertRaises(Exception) as context:
+            status, creds, queue_url, tile_index_queue_url, tile_bucket, params, tile_count = b.join(23)
+
+            self.assertTrue('After {} attempts, failed to join ingest job: {}'.format(50, ERROR_TEXT))
```

Comment:
This assert is incorrect and also unnecessary since you’re expecting an exception.

Reply:
The assertTrue is inside the "catch from with assertRaises.  It worked when I ran it.

Reply:
I guess I am not comparing anything am I.  I should be a comparison.

Reply:
The test passes because the exception is raised before the assert True. `assertTrue` on a string will always fail.